ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -117,6 +118,49 @@ end = struct
     mode := next_mode
 
   let current_mode () = !mode
+
+  let is_ipv6_enabled () = !ipv6_enabled
+end
+
+module Client_certificate_auth_server = struct
+  let management_server = ref None
+
+  let must_be_running ~__context ~mgmt_enabled =
+    let pool = Helpers.get_pool ~__context in
+    mgmt_enabled
+    && Pool_role.is_master ()
+    && Db.Pool.get_client_certificate_auth_enabled ~__context ~self:pool
+
+  let configure_port cmd =
+    let cmd' = if cmd then "open" else "close" in
+    let _ =
+      Helpers.call_script
+        !Xapi_globs.firewall_port_config_script
+        [cmd'; string_of_int Constants.https_port_clientcert]
+    in
+    ()
+
+  let start () =
```

Comment:
If IPv6 was enabled/disabled whilst xapi was running, I suspect we'd need something more in this function. However, I don't even know if that is possible, and it is presumably not supported anyway

Reply:
This part is independent from IP addresses. It combines management of the unix socket (no TCP/IP) and the TCP port entry in the firewall (independent of IP addressing).